Support creating project in current directory

### What problem does this feature solve?
Makes a step towards more easily using the 3.x CLI with other non-javascript backend projects such as asp.net or flask.

Currently, invoking `vue create .` in an empty directory throws the following error

```
🚀  Invoking generators...
 ERROR  Error: Cannot find module '@vue/cli-service/generator' from '/Users/cecilphillip/Desktop/toss'
Error: Cannot find module '@vue/cli-service/generator' from '/Users/cecilphillip/Desktop/toss'
    at Function.module.exports [as sync] (/usr/local/lib/node_modules/@vue/cli/node_modules/resolve/lib/sync.js:40:15)
    at Object.keys.map.id (/usr/local/lib/node_modules/@vue/cli/lib/Creator.js:246:30)
    at Array.map (<anonymous>)
    at Creator.resolvePlugins (/usr/local/lib/node_modules/@vue/cli/lib/Creator.js:245:36)
    at Creator.create (/usr/local/lib/node_modules/@vue/cli/lib/Creator.js:134:26)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:160:7)
```

### What does the proposed API look like?
`vue create .`   ||   `vue create --here`

<!-- generated by vue-issues. DO NOT REMOVE -->

--- a/packages/@vue/cli/lib/create.js
+++ b/packages/@vue/cli/lib/create.js
@@ -8,28 +8,44 @@ const clearConsole = require('./util/clearConsole')
 const { error, stopSpinner } = require('@vue/cli-shared-utils')
 
 async function create (projectName, options) {
-  const targetDir = path.resolve(process.cwd(), projectName)
+  const inCurrent = projectName === '.'
+  const name = inCurrent ? path.relative('../', process.cwd()) : projectName
+  const targetDir = path.resolve(projectName || '.')
+
   if (fs.existsSync(targetDir)) {
     if (options.force) {
       rimraf.sync(targetDir)
     } else {
       await clearConsole()
-      const { action } = await inquirer.prompt([
-        {
-          name: 'action',
-          type: 'list',
-          message: `Target directory ${chalk.cyan(targetDir)} already exists. Pick an action:`,
-          choices: [
-            { name: 'Overwrite', value: 'overwrite' },
-            { name: 'Merge', value: 'merge' },
-            { name: 'Cancel', value: false }
-          ]
+      if (inCurrent) {
+        const { ok } = await inquirer.prompt([
+          {
+            name: 'ok',
+            type: 'confirm',
+            message: `Generate project in current directory ${chalk.cyan(targetDir)} ?`
+          }
+        ])
+        if (!ok) {
+          return
         }
-      ])
-      if (!action) {
-        return
-      } else if (action === 'overwrite') {
-        rimraf.sync(targetDir)
+      } else {
+        const { action } = await inquirer.prompt([
+          {
+            name: 'action',
+            type: 'list',
+            message: `Target directory ${chalk.cyan(targetDir)} already exists. Pick an action:`,
+            choices: [
+              { name: 'Overwrite', value: 'overwrite' },
+              { name: 'Merge', value: 'merge' },
+              { name: 'Cancel', value: false }
+            ]
+          }
+        ])
+        if (!action) {
+          return
+        } else if (action === 'overwrite') {
+          rimraf.sync(targetDir)
+        }
       }
     }
   }
@@ -46,7 +62,7 @@ async function create (projectName, options) {
     'e2e'
   ].map(file => require(`./promptModules/${file}`))
 
-  const creator = new Creator(projectName, targetDir, promptModules)
+  const creator = new Creator(name, targetDir, promptModules)
   await creator.create(options)
 }
 

